### PR TITLE
[ST-1046] Use Prettier

### PR DESCRIPTION
**Pull request by jherdman**
Death to bike shedding.

### CODE REVIEW

File: .prettierignore [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+/coverage/
+/libpeerconnection.log
+/npm-debug.log*
+/testem.log
+/yarn-error.log
+
+# ember-try
+/.node_modules.ember-try/
+/bower.json.ember-try
+/package.json.ember-try
```
Comment (jherdman) [inline .prettierignore:26]: The bulk of this is based upon the recommendation to use one's `.gitignore` file as the basis.

**Review by ghedamat — APPROVED** [review]
i'm fine w/this but i wonder how it plays with our eslint config

also 

should we extend the tests so that we check that it passes prettier before merge/

it will take a bit to get devs to use it

especially until it can only be enabled on certain projects

**Comment by jherdman** [comment]
> i'm fine w/this but i wonder how it plays with our eslint config

IMHO we kill our config and just move to Prettier. It's probably best in the long run.

> should we extend the tests so that we check that it passes prettier before merge/

The setup I've instituted here automagically applies Prettier before you commit. It's pretty great! More details here: https://prettier.io/docs/en/precommit.html#option-1-lint-stagedhttpsgithubcomokonetlint-staged

File: .eslintrc.js [hunk 2]
```diff
@@ -7,21 +7,16 @@ module.exports = {
...
   },
-  plugins: [
-    'ember'
-  ],
-  extends: [
-    'eslint:recommended',
-    'plugin:ember/recommended'
-  ],
+  plugins: ['ember'],
+  extends: ['eslint:recommended', 'plugin:ember/recommended', 'prettier'],
```
Comment (ivanvolti) [inline .eslintrc.js:14]: [eslint-config-prettier](https://github.com/prettier/eslint-config-prettier) will turn off some ESLint rules. As a result, some of our code conventions that were enforced by ESLint will not be enforced anymore. We could fix that by using Prettier as a linter but that will require to run `prettier --write` for all our .js files.

**Review by ivanvolti — APPROVED** [review]
Looks good.👍

File: .eslintrc.js (hunk 2 above)
Comment (jherdman) [reply .eslintrc.js:14]: I'll dig into this.

**Review by ghedamat — APPROVED** [review]

**Comment by ghedamat** [comment]
@jherdman is there something to do here?

**Comment by jherdman** [comment]
@ghedamat yeah, I'll polish it up and open for review this morning.

**Pull request merged**